src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryWorker.cs

```diff
@@ -204,6 +192,50 @@ public void Validate([NotNull] HttpContext httpContext, string cookieToken, stri
                 deserializedFormToken);
         }
 
+        // [ ENTRY POINT ]
+        // Generates an anti-XSRF cookie token for the current user.
+        public void GetCookieTokenAndHeader([NotNull] HttpContext httpContext)
+        {
+            CheckSSLConfig(httpContext);
+
+            var oldCookieToken = GetCookieTokenNoThrow(httpContext);
+            AntiForgeryToken newCookieToken = ValidateAndGenerateNewToken(oldCookieToken);
+            if (newCookieToken != null)
+            {
+                SaveCookieTokenAndHeader(httpContext, newCookieToken);
+            }
+        }
+
+        private AntiForgeryToken ValidateAndGenerateNewToken(AntiForgeryToken oldCookieToken)
+        {
+            AntiForgeryToken newCookieToken = null;
```

Comment:
ah we are doing a null check, please add a comment to the method saying that this method returns null if the old cookie is still valid.